x-pack/plugins/watcher/server/routes/api/watches/register_delete_route.ts

```diff
@@ -37,25 +43,21 @@ function deleteWatches(callWithRequest: any, watchIds: string[]) {
   });
 }
 
-export function registerDeleteRoute(deps: RouteDependencies) {
-  const handler: RequestHandler<any, any, any> = async (ctx, request, response) => {
-    const callWithRequest = callWithRequestFactory(deps.elasticsearchService, request);
-
-    try {
-      const results = await deleteWatches(callWithRequest, request.body.watchIds);
-      return response.ok({ body: { results } });
-    } catch (e) {
-      return response.internalError({ body: e });
-    }
-  };
+const handler: RequestHandler<unknown, unknown, BodySchema> = async (ctx, request, response) => {
```

Comment:
nit: `void` seems better than `unknown`

Reply:
Latest version does not have the handler exported, but I just wanted to say that I think this can be a semantic. `unknown` to me more accurately describes the values in `params`, `body` or `query` when they haven't been validated. Technically, platform passes through an empty object but that is an implementation detail. So viewing that value as `unknown` is safest - for instance, if you do `if (request.body) {` (weird example 🎉) check and you haven't defined a validation for body, it will be true where `void` may create the expectation that it is `undefined` and the dev may expect that statement to evaluate to false.

Reply:
I am not sure I follow you. When we say that the body type is  `void` it means that there should not be any body right? And to reject the request if a body is provided. Or am I missing something?
`unknown` means: "body is allowed, we just don't know in what shape it will come".

Reply:
> And to reject the request if a body is provided

This is not the current behaviour. If you provide a body to a route without body validation, it does not reject, it just does not pass through the body of the request. It actually [passes through](https://github.com/elastic/kibana/blob/master/src/core/MIGRATION_EXAMPLES.md#http-routes) `{}`.

> unknown means: "body is allowed, we just don't know in what shape it will come".

This is where it is a difference in interpretation :). `unknown` conveys less information. It means treat this value as though we know nothing about it. I.e., it "could be anything" which is more encompassing than `void` which means "it is nothing".

Reply:
I see, so to be more precise it would be 

```js
const emptySchema = schema.never();
type Empty = TypeOf<typeof emptySchema>;

const handler: RequestHandler<Empty, Empty, BodySchema>
```

> it "could be anything, even nothing" 

But maybe we don't want it to be anything. We _want_ it to be **nothing** and reject

Reply:
That is an interesting idea which I hadn't considered! But yes, if you wanted that behaviour that's what you would need to add.